scheduler/src/main/java/org/cloudfoundry/autoscaler/scheduler/service/ScheduleManager.java

```diff
@@ -0,0 +1,399 @@
+import org.cloudfoundry.autoscaler.scheduler.util.DataValidationHelper;
+import org.cloudfoundry.autoscaler.scheduler.util.DateHelper;
+import org.cloudfoundry.autoscaler.scheduler.util.SpecificDateScheduleDateTime;
+import org.cloudfoundry.autoscaler.scheduler.util.ScheduleTypeEnum;
+import org.cloudfoundry.autoscaler.scheduler.util.error.DatabaseValidationException;
+import org.cloudfoundry.autoscaler.scheduler.util.error.SchedulerInternalException;
+import org.cloudfoundry.autoscaler.scheduler.util.error.ValidationErrorResult;
+import org.quartz.SchedulerException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+/**
+ * Service class to persist the schedule entity in the database and create
+ * scheduled job.
+ * 
+ * 
+ *
+ */
+@Service
+public class ScheduleManager {
+
+	@Autowired
+	private ScheduleDao scheduleDao;
+	@Autowired
+	private ScheduleJobManager scalingJobManager;
+	@Autowired
+	private ValidationErrorResult validationErrorResult;
+
+	private String scheduleBeingProcessed; // Specific date/Recurring
+	private Logger logger = LogManager.getLogger(this.getClass());
+
+	/**
+	 * Calls dao and fetch all the schedules for the specified application id.
+	 * 
+	 * @param appId
+	 * @return
+	 * @throws Exception
+	 */
+	public ApplicationScalingSchedules getAllSchedules(String appId) {
+		logger.info("Get All schedules for application: " + appId);
+		
+		List<ScheduleEntity> allScheduleEntitiesForApp = null;
+		ApplicationScalingSchedules applicationScalingSchedules = null;
+		try {
+			allScheduleEntitiesForApp = scheduleDao.findAllSchedulesByAppId(appId);
+		} catch (DatabaseValidationException dve) {
+
+			validationErrorResult.addErrorForDatabaseValidationException(dve, "schedule.database.error.get.failed",
+					"app_id=" + appId);
+			throw new SchedulerInternalException("Database error", dve);
+		}
+
+		if (allScheduleEntitiesForApp != null) {
+			applicationScalingSchedules = populateScheduleModel(allScheduleEntitiesForApp);
+		}
+		return applicationScalingSchedules;
+	}
+
+	/**
+	 * Helper method to extract the data from the schedule entity collection,
+	 * and populate schedule model with specific schedules and recurring
+	 * schedules.
+	 * 
+	 * @param allScheduleEntitiesForApp - List of all schedules
+	 * @return
+	 */
+	private ApplicationScalingSchedules populateScheduleModel(List<ScheduleEntity> allScheduleEntitiesForApp) {
+
+		ApplicationScalingSchedules applicationScalingSchedules = new ApplicationScalingSchedules();
+		List<ScheduleEntity> specificDateSchedules = new ArrayList<ScheduleEntity>();
+		applicationScalingSchedules.setSpecific_date(specificDateSchedules);
+
+		for (ScheduleEntity scheduleEntity : allScheduleEntitiesForApp) {
+			switch (ScheduleTypeEnum.getEnum(scheduleEntity.getScheduleType())) {
```

Comment:
Switch in this case might not be any faster than if else since small set of values, so will change it to if.